### ISSUE TITLE
Snapshot plugin versions are not supported

### ISSUE THREAD

**Issue by beary** [gradle/gradle#3902]
### Expected Behavior
Gradle report `Plugin Repositories (snapshot plugin versions are not supported)]`when use plugins DSL because of [here codes](https://github.com/gradle/gradle/blob/master/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/ArtifactRepositoriesPluginResolver.java#L72).
```groovy
/* build.gradle */
plugins {
    id 'org.springframework.boot' version '2.0.0.BUILD-SNAPSHOT'
}
```
I guess most people know what the `snapshot` means, so why don't allow use `snapshot` version plugin for more convenience when trying some new version things. 

### Your Environment
java 9
gradle 4.4.1
macOS 10.13.2

**Comment by mat013** [comment]
Seems like there has been a discussion for a long time: https://discuss.gradle.org/t/plugins-block-with-custom-repo-has-severe-limitations/18244/19

**Comment by sdeleuze** [comment]
This is very much needed on Spring side!

**Comment by sdeleuze** [comment]
FYI this issue is blocking start.spring.io to leverage the "new " `plugins { }` declaration and prevent us to support Gradle Kotlin DSL, see https://github.com/spring-io/initializr/issues/334#issuecomment-372254299 for more details.

**Comment by eskatos** [comment]
The fix in #5762 was merged to `master` and will be part of Gradle 4.10.
Thank you @sdeleuze for fixing this long standing issue!
